### DISCUSSION TITLE
Is there a way to use the full-site editing experience within a php template page?

### DISCUSSION THREAD

**Discussion by jakeparis** [WordPress/gutenberg#38542] [Q&A]
I'm thinking it would be ideal if we could continue to use our existing php templates, but replace the loop section with an area to insert various block templates. 

I can't see any way to do this. From what I can tell, you can _either_ use a completely FSE-built page _or_ the standard php template pages, but not both together. 

Something like this would be ideal, as our site headers and footers are very complex and would be quite hard to duplicate in blocks alone. However, we have a bunch of old-style page templates, most of which are just different layouts in the middle section. Using block-based layouts in the middle section, while still allowing php to render the header/footer could be ideal. 

Also, I'm brand new at the FSE experience, so any other ways to approach this are also appreciated. I'm sure I can't be the only one interested in FSE, but not wanting to completely give up our dynamic php sections.

**Comment by annezazu** [comment]
Great question! Exciting to hear you're digging in. From what I know, the current experience is a bit split with either PHP templates or block templates. You can see some discussion around this in the following issues: https://github.com/WordPress/gutenberg/issues/24803 https://github.com/WordPress/gutenberg/issues/29024

Right now, you can do things like [adding template editor support](https://developer.wordpress.org/themes/block-themes/converting-a-classic-theme-to-a-block-theme/#enabling-template-editor) while still relying on PHP templates in a classically built theme. You also might enjoy this issue that digs into a specific use case around allowing usage of block template parts without fully using a block template (kind of the opposite of what you're describing with keeping header/footer in place): https://github.com/WordPress/gutenberg/issues/37943

Some work remains to build out the UI for this idea though but the underlying technical bits are there, which is why there is a more stark split. Hope this helps :)

**Comment by catkit** [comment]
@jakeparis You can call an html template part from within a php template using the `block_template_part()` function. The html template part can be edited from the FSE. The documentation can be found here: https://developer.wordpress.org/reference/functions/block_template_part/
